daemon/graphdriver/overlay/overlay.go

```diff
@@ -263,6 +300,20 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		}
 	}()
 
+	driver := &Driver{}
+	if opts != nil {
+		if err := d.parseStorageOpt(opts.StorageOpt, driver); err != nil {
+			return err
+		}
+	}
+
+	if driver.options.quota.Size > 0 && !strings.HasSuffix(dir, "-init") {
+		// enforce container disk project quota
+		if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
```

Comment:
As mentioned below, this only needs to be set on an `upper` directory to match the behavior of the overlay2 quota.